packages/visual-editor/src/components/DraggableHelpers/DragWrapper.tsx

```diff
@@ -0,0 +1,130 @@
+  'data-rfd-drag-handle-draggable-id'?: string;
+  'data-rfd-drag-handle-context-id'?: string;
+}
+
+export type WrapperTags = keyof Pick<
+  JSX.IntrinsicElements,
+  | 'div'
+  | 'span'
+  | 'section'
+  | 'article'
+  | 'aside'
+  | 'p'
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
+  | 'header'
+  | 'footer'
+  | 'nav'
+  | 'main'
+>;
+
+export interface DragWrapperProps
+  extends DraggableProvidedDragHandleProps,
+    DraggableProvidedDraggableProps,
+    HTMLAttributes<HTMLElement>,
+    React.PropsWithChildren {
+  'data-cf-node-id'?: string;
+  'data-ctfl-draggable-id'?: string;
+  'data-test-id'?: string;
+  'data-cf-node-block-id'?: string;
+  'data-cf-node-block-type'?: string;
+  innerRef?: (refNode: HTMLElement) => void;
+  wrapComponent?: boolean;
+  Tag?: WrapperTags;
+  ToolTipAndPlaceholder?: React.ReactNode;
+}
+
+export const DragWrapper: React.FC<DragWrapperProps> = ({
+  className,
+  children,
+  innerRef,
+  Tag = 'div',
+  ToolTipAndPlaceholder,
+  wrapComponent,
+  ...props
+}) => {
+  const wrapperRef = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (wrapComponent && wrapperRef.current) {
+      const componentElement = wrapperRef.current.querySelector(
+        ':not([data-tooltip])',
+      ) as HTMLElement;
+
+      if (!componentElement) {
+        return;
+      }
+
+      let width = componentElement.style.getPropertyValue('width');
+      let maxWidth = componentElement.style.getPropertyValue('max-width');
+
+      if (!width || !maxWidth) {
+        const styleSheets = Array.from(document.styleSheets);
+        for (let i = styleSheets.length - 1; i >= 0; i--) {
+          const cssRules = Array.from(styleSheets[i].cssRules || styleSheets[i].rules || []);
+          cssRules.some((cssRule) => {
+            if (cssRule instanceof CSSStyleRule) {
+              const selectorText = cssRule.selectorText as string;
+              if (
+                !selectorText ||
+                selectorText.startsWith('.styles-') ||
+                selectorText.startsWith('.contentful-container')
+              ) {
+                return;
+              }
+              if (componentElement?.matches(selectorText)) {
+                width = width || cssRule.style.width;
+                maxWidth = maxWidth || cssRule.style.maxWidth;
+                return !!width && !!maxWidth;
+              }
+            }
+          });
+
+          if (width && maxWidth) {
+            break;
+          }
+        }
+      }
+
+      if (width || maxWidth) {
+        // wrapperRef.current.style.setProperty('display', 'inline-block');
+        wrapperRef.current.style.setProperty('width', width);
+        wrapperRef.current.style.setProperty('max-width', maxWidth);
+        // wrapperRef.current.style.setProperty('height', '33px');
+        // wrapperRef.current.classList.add(...classes);
+      } else {
+        wrapperRef.current.style.setProperty('width', 'max-content');
+      }
```

Comment:
I noticed the "Nested Slots" example component isn't rendering full-width any more. It looks like the DraggableComponent wrapper is getting `width: max-content` from this logic.  I tried adding `width: 100%` style to the Nested Slots component's `styles.container` but it still didn't go full-width 🤔 

![Screenshot 2024-06-10 at 9 26 08 AM](https://github.com/contentful/experience-builder/assets/8539634/a69fc78e-21ce-494e-a620-e934222bfc1b)

Reply:
Refactored how styling works here and feels much better now